[Bug]: Fail to run Mirror Maker 2 connector task when using OAUTH to connect to kafka clusters

### Bug Description

I applied Mirror Maker 2 cr, and the pod is running successfully. But I checked the status of this new connector via rest api, 
I got "caused by: java.lang.IllegalArgumentException: Value not specified for key 'XXXX(**mm2 cluster alias**)' in JAAS config"

My authentication type is "oauth", and set trustedCertificates for TLS connection to the OAuth server.

### Steps to reproduce

_No response_

### Expected behavior

_No response_

### Strimzi version

0.39.0

### Kubernetes version

K8s 1.27

### Installation method

_No response_

### Infrastructure

_No response_

### Configuration files and logs

_No response_

### Additional context

I think I find the root cause of this issue, I will create a PR, and fix it.

Operator version: 0.39.0
MM2 CR yaml:
```
apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaMirrorMaker2
metadata:
  name: kafka-mirror-maker-2
spec:
  clusters:
  - alias: "strimzi-poc-e1"
    authentication:
      type: tls
      certificateAndKey:
        secretName: strimzi-poc-kafka-bootstrap
        certificate: tls.crt
        key: tls.key
    bootstrapServers: strimzi-poc-kafka-bootstrap:9093
    tls:
      trustedCertificates:
      - certificate: ca.crt
        secretName: strimzi-poc-kafka-bootstrap
  - alias: "strimzi-poc-e2"
    authentication:
      clientId: xxxxxxx
      clientSecret: 
        key: client_secret
        secretName: jwt-client-secret
      tokenEndpointUri: https://xxxxxxxxx
# We can reproduce this issue when we enabled tlsTrustedCertificates for oauth
      tlsTrustedCertificates:
      - certificate: ca.crt
         secretName: strimzi-poc-kafka-bootstrap
      type: oauth
    bootstrapServers: XXXXXXXXXXXXX:443
    config:
      config.storage.replication.factor: -1
      offset.storage.replication.factor: -1
      status.storage.replication.factor: -1
    tls:
      trustedCertificates:
      - certificate: ca.crt
        secretName: strimzi-poc-kafka-bootstrap
  connectCluster: "strimzi-poc-e1"
  mirrors:
  - checkpointConnector:
      config:
        checkpoints.topic.replication.factor: 3
    groupsPattern: .*
    heartbeatConnector:
      config:
        heartbeats.topic.replication.factor: 3
    sourceCluster: "strimzi-poc-e2"
    sourceConnector:
      config:
        offset-syncs.topic.replication.factor: 3
        replication.factor: 3
    targetCluster: "strimzi-poc-e1"
    topicsPattern: .*
  replicas: 1
  resources:
    limits:
      cpu: 1000m
      memory: 2Gi
    requests:
      cpu: 1000m
      memory: 2Gi
  template:
    connectContainer:
      securityContext:
        allowPrivilegeEscalation: false
        readOnlyRootFilesystem: true
        runAsUser: 1001
        runAsGroup: 1001
        runAsNonRoot: true
        privileged: false
        capabilities:
          drop:
          - ALL
        seccompProfile:
          type: RuntimeDefault
    pod:
      metadata:
        labels:
          app: mm2
          applicationid: ap141882
      topologySpreadConstraints:
      - labelSelector:
          matchExpressions:
          - key: app
            operator: In
            values:
            - mm2
        maxSkew: 1
        topologyKey: topology.kubernetes.io/zone
        whenUnsatisfiable: DoNotSchedule
  version: 3.6.1
```
MM2 connector config(generated by operator):
GET /connectors/strimzi-poc-e2->strimzi-poc-e1.MirrorSourceConnector/config
```
{
	"connector.class": "org.apache.kafka.connect.mirror.MirrorSourceConnector",
	"replication.factor": "3",
	"source.cluster.ssl.truststore.location": "/tmp/kafka/clusters/strimzi-poc-e2.truststore.p12",
	"source.cluster.ssl.truststore.type": "PKCS12",
	"source.cluster.alias": "strimzi-poc-e2",
	"target.cluster.ssl.truststore.location": "/tmp/kafka/clusters/strimzi-poc-e1.truststore.p12",
	"source.cluster.sasl.login.callback.handler.class": "io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler",
	"source.cluster.config.storage.replication.factor": "-1",
	"target.cluster.ssl.truststore.type": "PKCS12",
	"target.cluster.security.protocol": "SSL",
	"target.cluster.ssl.keystore.location": "/tmp/kafka/clusters/strimzi-poc-e1.keystore.p12",
	"consumer.client.rack": "${file:/tmp/strimzi-connect.properties:consumer.client.rack}",
	"offset-syncs.topic.replication.factor": "3",
	"topics": ".*",
	"target.cluster.ssl.keystore.password": "${file:/tmp/strimzi-mirrormaker2-connector.properties:ssl.keystore.password}",
	"groups": ".*",
	"target.cluster.ssl.truststore.password": "${file:/tmp/strimzi-mirrormaker2-connector.properties:ssl.truststore.password}",
	"source.cluster.status.storage.replication.factor": "-1",
	"source.cluster.sasl.jaas.config": "org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required oauth.token.endpoint.uri=\"https://XXXXXXXXX\" oauth.client.secret=\"${file:/tmp/strimzi-mirrormaker2-connector.properties:strimzi-poc-e2.oauth.client.secret}\" oauth.ssl.truststore.location=\"/tmp/kafka/clusters/\"strimzi-poc-e2\"-oauth.truststore.p12\" oauth.ssl.truststore.type=\"PKCS12\" oauth.ssl.truststore.password=\"\"${file:/tmp/strimzi-mirrormaker2-connector.properties:oauth.ssl.truststore.password}\"\" oauth.client.id=\"XXXXXXXX\";",
	"source.cluster.bootstrap.servers": "XXXXXXXX:443",
	"source.cluster.sasl.mechanism": "OAUTHBEARER",
	"target.cluster.alias": "strimzi-poc-e1",
	"target.cluster.ssl.keystore.type": "PKCS12",
	"name": "strimzi-poc-e2->strimzi-poc-e1.MirrorSourceConnector",
	"target.cluster.bootstrap.servers": "strimzi-poc-kafka-bootstrap:9093",
	"source.cluster.offset.storage.replication.factor": "-1",
	"source.cluster.ssl.truststore.password": "${file:/tmp/strimzi-mirrormaker2-connector.properties:ssl.truststore.password}",
	"source.cluster.security.protocol": "SASL_SSL"
}
```
You can see below jaas config from above json response, there are unnecessary two double quotes(/\"strimzi-poc-e2\"-oauth) on this trutstore file path.
```
oauth.ssl.truststore.location=\"/tmp/kafka/clusters/\"strimzi-poc-e2\"-oauth.truststore.p12\"
```
GET /connectors/strimzi-poc-e2->strimzi-poc-e1.MirrorSourceConnector/status
Error response:
```
{
	"name": "strimzi-poc-e2->strimzi-poc-e1.MirrorSourceConnector",
	"connector": {
		"state": "FAILED",
		"worker_id": "kafka-mirror-maker-2-mirrormaker2-0.kafka-mirror-maker-2-mirrormaker2.strimzi-kafka-dev.svc:8083",
		"trace": "org.apache.kafka.common.KafkaException: The constructor of org.apache.kafka.clients.admin.ForwardingAdmin threw an exception\n\tat org.apache.kafka.common.utils.Utils.newParameterizedInstance(Utils.java:471)\n\tat org.apache.kafka.connect.mirror.MirrorConnectorConfig.forwardingAdmin(MirrorConnectorConfig.java:231)\n\tat org.apache.kafka.connect.mirror.MirrorSourceConnector.start(MirrorSourceConnector.java:160)\n\tat org.apache.kafka.connect.runtime.WorkerConnector.doStart(WorkerConnector.java:192)\n\tat org.apache.kafka.connect.runtime.WorkerConnector.start(WorkerConnector.java:217)\n\tat org.apache.kafka.connect.runtime.WorkerConnector.doTransitionTo(WorkerConnector.java:376)\n\tat org.apache.kafka.connect.runtime.WorkerConnector.doTransitionTo(WorkerConnector.java:357)\n\tat org.apache.kafka.connect.runtime.WorkerConnector.doRun(WorkerConnector.java:144)\n\tat org.apache.kafka.connect.runtime.WorkerConnector.run(WorkerConnector.java:122)\n\tat org.apache.kafka.connect.runtime.isolation.Plugins.lambda$withClassLoader$1(Plugins.java:236)\n\tat java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)\n\tat java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:840)\nCaused by: org.apache.kafka.common.KafkaException: Failed to create new KafkaAdminClient\n\tat org.apache.kafka.clients.admin.KafkaAdminClient.createInternal(KafkaAdminClient.java:519)\n\tat org.apache.kafka.clients.admin.Admin.create(Admin.java:144)\n\tat org.apache.kafka.clients.admin.ForwardingAdmin.<init>(ForwardingAdmin.java:51)\n\tat java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)\n\tat java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)\n\tat java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)\n\tat java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)\n\tat java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)\n\tat org.apache.kafka.common.utils.Utils.newParameterizedInstance(Utils.java:460)\n\t... 14 more\nCaused by: org.apache.kafka.common.KafkaException: Failed to create new NetworkClient\n\tat org.apache.kafka.clients.ClientUtils.createNetworkClient(ClientUtils.java:245)\n\tat org.apache.kafka.clients.ClientUtils.createNetworkClient(ClientUtils.java:185)\n\tat org.apache.kafka.clients.admin.KafkaAdminClient.createInternal(KafkaAdminClient.java:503)\n\t... 22 more\nCaused by: java.lang.IllegalArgumentException: Value not specified for key 'strimzi-poc-e2' in JAAS config\n\tat org.apache.kafka.common.security.JaasConfig.parseAppConfigurationEntry(JaasConfig.java:116)\n\tat org.apache.kafka.common.security.JaasConfig.<init>(JaasConfig.java:63)\n\tat org.apache.kafka.common.security.JaasContext.load(JaasContext.java:93)\n\tat org.apache.kafka.common.security.JaasContext.loadClientContext(JaasContext.java:87)\n\tat org.apache.kafka.common.network.ChannelBuilders.create(ChannelBuilders.java:167)\n\tat org.apache.kafka.common.network.ChannelBuilders.clientChannelBuilder(ChannelBuilders.java:81)\n\tat org.apache.kafka.clients.ClientUtils.createChannelBuilder(ClientUtils.java:117)\n\tat org.apache.kafka.clients.ClientUtils.createNetworkClient(ClientUtils.java:217)\n\t... 24 more\n"
	},
	"tasks": [],
	"type": "source"
}
```
You can focus on "Caused by: java.lang.IllegalArgumentException: Value not specified for key 'strimzi-poc-e2' in JAAS config\n\tat " from above json response.

--- a/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaMirrorMaker2Connectors.java
+++ b/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaMirrorMaker2Connectors.java
@@ -282,8 +282,8 @@ public class KafkaMirrorMaker2Connectors {
         }
 
         if (oauth.getTlsTrustedCertificates() != null && !oauth.getTlsTrustedCertificates().isEmpty()) {
-            jaasOptions.put("oauth.ssl.truststore.location", "/tmp/kafka/clusters/\"" + cluster.getAlias() + "\"-oauth.truststore.p12");
-            jaasOptions.put("oauth.ssl.truststore.password", "\"${file:" + CONNECTORS_CONFIG_FILE + ":oauth.ssl.truststore.password}\"");
+            jaasOptions.put("oauth.ssl.truststore.location", "/tmp/kafka/clusters/" + cluster.getAlias() + "-oauth.truststore.p12");
+            jaasOptions.put("oauth.ssl.truststore.password", "${file:" + CONNECTORS_CONFIG_FILE + ":oauth.ssl.truststore.password}");
             jaasOptions.put("oauth.ssl.truststore.type", "PKCS12");
         }
 

--- a/cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaMirrorMaker2ConnectorsTest.java
+++ b/cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaMirrorMaker2ConnectorsTest.java
@@ -5,6 +5,7 @@
 package io.strimzi.operator.cluster.model;
 
 import io.strimzi.api.kafka.model.common.CertAndKeySecretSourceBuilder;
+import io.strimzi.api.kafka.model.common.CertSecretSourceBuilder;
 import io.strimzi.api.kafka.model.common.ConnectorState;
 import io.strimzi.api.kafka.model.connector.KafkaConnector;
 import io.strimzi.api.kafka.model.mirrormaker2.KafkaMirrorMaker2;
@@ -615,6 +616,7 @@ public class KafkaMirrorMaker2ConnectorsTest {
                         .withKey("refreshTokenKey")
                         .withSecretName("refreshTokenSecretName")
                     .endRefreshToken()
+                    .withTlsTrustedCertificates(new CertSecretSourceBuilder().withCertificate("ca.crt").withSecretName("my-oauth-secret").build())
                 .endKafkaClientAuthenticationOAuth()
                 .build();
 
@@ -627,7 +629,10 @@ public class KafkaMirrorMaker2ConnectorsTest {
                 is(Map.of("oauth.client.secret", "${file:/tmp/strimzi-mirrormaker2-connector.properties:sourceClusterAlias.oauth.client.secret}",
                         "oauth.access.token", "${file:/tmp/strimzi-mirrormaker2-connector.properties:sourceClusterAlias.oauth.access.token}",
                         "oauth.refresh.token", "${file:/tmp/strimzi-mirrormaker2-connector.properties:sourceClusterAlias.oauth.refresh.token}",
-                        "oauth.password.grant.password", "${file:/tmp/strimzi-mirrormaker2-connector.properties:sourceClusterAlias.oauth.password.grant.password}")));
+                        "oauth.password.grant.password", "${file:/tmp/strimzi-mirrormaker2-connector.properties:sourceClusterAlias.oauth.password.grant.password}",
+                        "oauth.ssl.truststore.location", "/tmp/kafka/clusters/sourceClusterAlias-oauth.truststore.p12",
+                        "oauth.ssl.truststore.type", "PKCS12",
+                        "oauth.ssl.truststore.password", "${file:/tmp/strimzi-mirrormaker2-connector.properties:oauth.ssl.truststore.password}")));
 
         assertThat(config,
                 is(Map.of("prefix.alias", "sourceClusterAlias",

--- a/systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthTlsST.java
+++ b/systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthTlsST.java
@@ -13,6 +13,9 @@ import io.strimzi.api.kafka.model.kafka.KafkaResources;
 import io.strimzi.api.kafka.model.kafka.listener.GenericKafkaListenerBuilder;
 import io.strimzi.api.kafka.model.kafka.listener.KafkaListenerType;
 import io.strimzi.api.kafka.model.mirrormaker.KafkaMirrorMakerResources;
+import io.strimzi.api.kafka.model.mirrormaker2.KafkaMirrorMaker2;
+import io.strimzi.api.kafka.model.mirrormaker2.KafkaMirrorMaker2ClusterSpec;
+import io.strimzi.api.kafka.model.mirrormaker2.KafkaMirrorMaker2ClusterSpecBuilder;
 import io.strimzi.operator.common.model.Labels;
 import io.strimzi.systemtest.Environment;
 import io.strimzi.systemtest.TestConstants;
@@ -31,6 +34,7 @@ import io.strimzi.systemtest.resources.kubernetes.NetworkPolicyResource;
 import io.strimzi.systemtest.storage.TestStorage;
 import io.strimzi.systemtest.templates.crd.KafkaBridgeTemplates;
 import io.strimzi.systemtest.templates.crd.KafkaConnectTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaMirrorMaker2Templates;
 import io.strimzi.systemtest.templates.crd.KafkaMirrorMakerTemplates;
 import io.strimzi.systemtest.templates.crd.KafkaNodePoolTemplates;
 import io.strimzi.systemtest.templates.crd.KafkaTemplates;
@@ -41,13 +45,20 @@ import io.strimzi.systemtest.utils.ClientUtils;
 import io.strimzi.systemtest.utils.kafkaUtils.KafkaConnectUtils;
 import io.strimzi.systemtest.utils.kafkaUtils.KafkaConnectorUtils;
 import io.strimzi.systemtest.utils.kafkaUtils.KafkaUserUtils;
+import io.strimzi.systemtest.utils.kubeUtils.controllers.JobUtils;
+import io.strimzi.test.TestUtils;
+import io.strimzi.test.WaitException;
+import io.strimzi.test.k8s.KubeClusterResource;
 import io.vertx.core.cli.annotations.Description;
 import org.apache.kafka.clients.consumer.ConsumerConfig;
 import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.logging.log4j.Level;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Tag;
+
+import java.time.Duration;
 
 import static io.strimzi.systemtest.TestConstants.ACCEPTANCE;
 import static io.strimzi.systemtest.TestConstants.ARM64_UNSUPPORTED;
@@ -410,6 +421,187 @@ public class OauthTlsST extends OauthAbstractST {
         resourceManager.createResourceWithWait(kafkaOauthClientJob.consumerStrimziOauthTls(targetKafkaCluster));
 
         ClientUtils.waitForClientSuccess(consumerName, Environment.TEST_SUITE_NAMESPACE, testStorage.getMessageCount());
+    }
+
+    @Description("As a OAuth MirrorMaker 2, I am able to replicate Topic data using using encrypted communication")
+    @IsolatedTest("Using more tha one Kafka cluster in one Namespace")
+    @Tag(MIRROR_MAKER)
+    @Tag(NODEPORT_SUPPORTED)
+    @SuppressWarnings({"checkstyle:MethodLength"})
+    void testMirrorMaker2() {
+        // Nodeport needs cluster wide rights to work properly which is not possible with STRIMZI_RBAC_SCOPE=NAMESPACE
+        assumeFalse(Environment.isNamespaceRbacScope());
+        final TestStorage testStorage = new TestStorage(ResourceManager.getTestContext());
+
+        String producerName = OAUTH_PRODUCER_NAME + "-" + testStorage.getClusterName();
+        String consumerName = OAUTH_CONSUMER_NAME + "-" + testStorage.getClusterName();
+
+        resourceManager.createResourceWithWait(KafkaTopicTemplates.topic(oauthClusterName, testStorage.getTopicName(), Environment.TEST_SUITE_NAMESPACE).build());
+
+        KafkaOauthClients oauthExampleClients = new KafkaOauthClientsBuilder()
+            .withNamespaceName(Environment.TEST_SUITE_NAMESPACE)
+            .withProducerName(producerName)
+            .withConsumerName(consumerName)
+            .withBootstrapAddress(KafkaResources.tlsBootstrapAddress(oauthClusterName))
+            .withTopicName(testStorage.getTopicName())
+            .withMessageCount(testStorage.getMessageCount())
+            .withOauthClientId(OAUTH_CLIENT_NAME)
+            .withOauthClientSecret(OAUTH_CLIENT_SECRET)
+            .withOauthTokenEndpointUri(keycloakInstance.getOauthTokenEndpointUri())
+            .build();
+
+        resourceManager.createResourceWithWait(oauthExampleClients.producerStrimziOauthTls(oauthClusterName));
+        ClientUtils.waitForClientSuccess(producerName, Environment.TEST_SUITE_NAMESPACE, testStorage.getMessageCount());
+
+        resourceManager.createResourceWithWait(oauthExampleClients.consumerStrimziOauthTls(oauthClusterName));
+        ClientUtils.waitForClientSuccess(consumerName, Environment.TEST_SUITE_NAMESPACE, testStorage.getMessageCount());
+
+        String targetKafkaCluster = oauthClusterName + "-target";
+        String kafkaSourceClusterName = oauthClusterName;
+
+        resourceManager.createResourceWithWait(
+            NodePoolsConverter.convertNodePoolsIfNeeded(
+                KafkaNodePoolTemplates.brokerPool(testStorage.getNamespaceName(), testStorage.getTargetBrokerPoolName(), targetKafkaCluster, 1).build(),
+                KafkaNodePoolTemplates.controllerPool(testStorage.getNamespaceName(), testStorage.getTargetControllerPoolName(), targetKafkaCluster, 1).build()
+            )
+        );
+        resourceManager.createResourceWithWait(KafkaTemplates.kafkaEphemeral(targetKafkaCluster, 1, 1)
+            .editMetadata()
+                .withNamespace(Environment.TEST_SUITE_NAMESPACE)
+            .endMetadata()
+            .editSpec()
+                .editKafka()
+                    .withListeners(OauthAbstractST.BUILD_OAUTH_TLS_LISTENER.apply(keycloakInstance),
+                            new GenericKafkaListenerBuilder()
+                                .withName(TestConstants.EXTERNAL_LISTENER_DEFAULT_NAME)
+                                .withPort(9094)
+                                .withType(KafkaListenerType.NODEPORT)
+                                .withTls(true)
+                                .withNewKafkaListenerAuthenticationOAuth()
+                                .withValidIssuerUri(keycloakInstance.getValidIssuerUri())
+                                .withJwksExpirySeconds(keycloakInstance.getJwksExpireSeconds())
+                                .withJwksRefreshSeconds(keycloakInstance.getJwksRefreshSeconds())
+                                .withJwksEndpointUri(keycloakInstance.getJwksEndpointUri())
+                                .withUserNameClaim(keycloakInstance.getUserNameClaim())
+                                .withTlsTrustedCertificates(
+                                    new CertSecretSourceBuilder()
+                                        .withSecretName(KeycloakInstance.KEYCLOAK_SECRET_NAME)
+                                        .withCertificate(KeycloakInstance.KEYCLOAK_SECRET_CERT)
+                                        .build())
+                                    .withDisableTlsHostnameVerification(true)
+                                .endKafkaListenerAuthenticationOAuth()
+                                .build())
+                .endKafka()
+            .endSpec()
+            .build());
+
+        // Deploy MirrorMaker2 with OAuth
+        KafkaMirrorMaker2ClusterSpec sourceClusterWithOauth = new KafkaMirrorMaker2ClusterSpecBuilder()
+            .withAlias(kafkaSourceClusterName)
+            .withConfig(connectorConfig)
+            .withBootstrapServers(KafkaResources.tlsBootstrapAddress(oauthClusterName))
+            // this is for kafka tls connection
+            .withNewTls()
+                .withTrustedCertificates(new CertSecretSourceBuilder()
+                    .withCertificate("ca.crt")
+                    .withSecretName(KafkaResources.clusterCaCertificateSecretName(oauthClusterName))
+                    .build())
+            .endTls()
+            .withNewKafkaClientAuthenticationOAuth()
+                .withTokenEndpointUri(keycloakInstance.getOauthTokenEndpointUri())
+                .withClientId("kafka-mirror-maker-2")
+                .withNewClientSecret()
+                    .withSecretName(MIRROR_MAKER_2_OAUTH_SECRET)
+                    .withKey(OAUTH_KEY)
+                .endClientSecret()
+                .withConnectTimeoutSeconds(CONNECT_TIMEOUT_S)
+                .withReadTimeoutSeconds(READ_TIMEOUT_S)
+                // this is for authorization server tls connection
+                .withTlsTrustedCertificates(new CertSecretSourceBuilder()
+                    .withSecretName(KeycloakInstance.KEYCLOAK_SECRET_NAME)
+                    .withCertificate(KeycloakInstance.KEYCLOAK_SECRET_CERT)
+                    .build())
+                .withDisableTlsHostnameVerification(true)
+            .endKafkaClientAuthenticationOAuth()
+            .build();
+
+        KafkaMirrorMaker2ClusterSpec targetClusterWithOauth = new KafkaMirrorMaker2ClusterSpecBuilder()
+            .withAlias(testStorage.getTargetClusterName())
+            .withConfig(connectorConfig)
+            .withBootstrapServers(KafkaResources.tlsBootstrapAddress(targetKafkaCluster))
+            // this is for kafka tls connection
+            .withNewTls()
+                .withTrustedCertificates(new CertSecretSourceBuilder()
+                    .withCertificate("ca.crt")
+                    .withSecretName(KafkaResources.clusterCaCertificateSecretName(targetKafkaCluster))
+                    .build())
+            .endTls()
+            .withNewKafkaClientAuthenticationOAuth()
+                .withTokenEndpointUri(keycloakInstance.getOauthTokenEndpointUri())
+                .withClientId("kafka-mirror-maker-2")
+                .withNewClientSecret()
+                    .withSecretName(MIRROR_MAKER_2_OAUTH_SECRET)
+                    .withKey(OAUTH_KEY)
+                .endClientSecret()
+                .withConnectTimeoutSeconds(CONNECT_TIMEOUT_S)
+                .withReadTimeoutSeconds(READ_TIMEOUT_S)
+                // this is for authorization server tls connection
+                .withTlsTrustedCertificates(new CertSecretSourceBuilder()
+                    .withSecretName(KeycloakInstance.KEYCLOAK_SECRET_NAME)
+                    .withCertificate(KeycloakInstance.KEYCLOAK_SECRET_CERT)
+                    .build())
+                .withDisableTlsHostnameVerification(true)
+            .endKafkaClientAuthenticationOAuth()
+            .build();
+
+        String kafkaTargetClusterTopicName = kafkaSourceClusterName + "." + testStorage.getTopicName();
+
+        resourceManager.createResourceWithWait(KafkaMirrorMaker2Templates.kafkaMirrorMaker2(oauthClusterName, testStorage.getTargetClusterName(), kafkaSourceClusterName, 1, false)
+            .editMetadata()
+                .withNamespace(Environment.TEST_SUITE_NAMESPACE)
+            .endMetadata()
+            .editSpec()
+                .withClusters(sourceClusterWithOauth, targetClusterWithOauth)
+                .editFirstMirror()
+                    .withSourceCluster(kafkaSourceClusterName)
+                .endMirror()
+            .endSpec()
+            .build());
+
+        final String kafkaMirrorMaker2PodName = kubeClient().listPods(Environment.TEST_SUITE_NAMESPACE, oauthClusterName, Labels.STRIMZI_KIND_LABEL, KafkaMirrorMaker2.RESOURCE_KIND).get(0).getMetadata().getName();
+        final String kafkaMirrorMaker2Logs = KubeClusterResource.cmdKubeClient(Environment.TEST_SUITE_NAMESPACE).execInCurrentNamespace(Level.DEBUG, "logs", kafkaMirrorMaker2PodName).out();
+        verifyOauthConfiguration(kafkaMirrorMaker2Logs);
+
+        TestUtils.waitFor("MirrorMaker2 to copy messages from " + kafkaSourceClusterName + " to " + testStorage.getTargetClusterName(),
+                Duration.ofSeconds(30).toMillis(), TestConstants.TIMEOUT_FOR_MIRROR_MAKER_COPY_MESSAGES_BETWEEN_BROKERS,
+                () -> {
+                    LOGGER.info("Deleting Job: {}/{}", Environment.TEST_SUITE_NAMESPACE, consumerName);
+                    JobUtils.deleteJobWithWait(Environment.TEST_SUITE_NAMESPACE, consumerName);
+
+                    LOGGER.info("Creating new client with new consumer-group and also to point on {} cluster", testStorage.getTargetClusterName());
+
+                    KafkaOauthClients kafkaOauthClientJob = new KafkaOauthClientsBuilder()
+                            .withNamespaceName(Environment.TEST_SUITE_NAMESPACE)
+                            .withProducerName(producerName)
+                            .withConsumerName(consumerName)
+                            .withBootstrapAddress(KafkaResources.tlsBootstrapAddress(targetKafkaCluster))
+                            .withTopicName(kafkaTargetClusterTopicName)
+                            .withMessageCount(testStorage.getMessageCount())
+                            .withOauthClientId(OAUTH_CLIENT_NAME)
+                            .withOauthClientSecret(OAUTH_CLIENT_SECRET)
+                            .withOauthTokenEndpointUri(keycloakInstance.getOauthTokenEndpointUri())
+                            .build();
+
+                    resourceManager.createResourceWithWait(kafkaOauthClientJob.consumerStrimziOauthTls(targetKafkaCluster));
+
+                    try {
+                        ClientUtils.waitForClientSuccess(consumerName, Environment.TEST_SUITE_NAMESPACE, testStorage.getMessageCount());
+                        return  true;
+                    } catch (WaitException e) {
+                        LOGGER.error("Failed while waiting for consumer to succeed", e);
+                        return false;
+                    }
+                });
     }
 
     @ParallelTest

--- a/systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthTlsST.java
+++ b/systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthTlsST.java
@@ -67,6 +67,7 @@ import static io.strimzi.systemtest.TestConstants.CONNECT;
 import static io.strimzi.systemtest.TestConstants.CONNECT_COMPONENTS;
 import static io.strimzi.systemtest.TestConstants.HTTP_BRIDGE_DEFAULT_PORT;
 import static io.strimzi.systemtest.TestConstants.MIRROR_MAKER;
+import static io.strimzi.systemtest.TestConstants.MIRROR_MAKER2;
 import static io.strimzi.systemtest.TestConstants.NODEPORT_SUPPORTED;
 import static io.strimzi.systemtest.TestConstants.OAUTH;
 import static io.strimzi.systemtest.TestConstants.REGRESSION;
@@ -425,7 +426,7 @@ public class OauthTlsST extends OauthAbstractST {
 
     @Description("As a OAuth MirrorMaker 2, I am able to replicate Topic data using using encrypted communication")
     @IsolatedTest("Using more tha one Kafka cluster in one Namespace")
-    @Tag(MIRROR_MAKER)
+    @Tag(MIRROR_MAKER2)
     @Tag(NODEPORT_SUPPORTED)
     @SuppressWarnings({"checkstyle:MethodLength"})
     void testMirrorMaker2() {

